app/models/turbo/streams/tag_builder.rb

```diff
@@ -27,6 +27,7 @@ class Turbo::Streams::TagBuilder
 
   def initialize(view_context)
     @view_context = view_context
+    @view_context.formats |= [:html]
```

Comment:
Will this also change things for the controller level tag builder? I think it's best we keep the no-fallback there, no? Unless well, I don't really remember why that's there.

We could add a `fallback_format:` to the builder intializer and then only include that in the view tag builder.

Reply:
The controller method just instantiates this class, so it would affect that too. 

I think it makes sense to have `format.turbo_stream` either call `turbo_stream` or render a `turbo_stream.erb` template. You wouldn't just not specify it if you wanted to use `html.erb` response. 👍

Reply:
Having the fallback keeps this inline with having an action with implicit rendering...

```ruby
def new
  @post = Post.new
end
```

Where if a TURBO_STREAM request comes in, the `format` is already set to `[:turbo_stream, :html]` so it would not only look for a turbo stream view first but also that format would be passed through to the view if it has any partcials.

Reply:
@excide3 Yeah, I get that point. Something irks me about altering the formats from within a `TagBuilder` though, especially since template rendering has been a rife source of security vulnerabilities in the past. So we do have to be a little careful here.

I'm leaning towards the block version you posted earlier 👍 @dhh may have a different take though 😄

Reply:
This works if we want to go that route:

```ruby
    when block_given?
      with_html_format do
        turbo_stream_action_tag name, target: target_name, template: @view_context.capture(&block)
      end
```
```ruby
    def with_html_format
      previous_formats = @view_context.formats
      @view_context.formats |= [:html]
      result = yield
      @view_context.formats = previous_formats
      result
    end
```

Reply:
We'd need an `ensure` for the reset to guard for exceptions and then we could just remove the `result` variable, but yes, that would be the other way. Let's go with the mutation for now and see how that goes!

Reply:
Ah yes, of course! As I was writing that I was like I know I'm missing something. 😜